test/plugin_functional/test_suites/core_plugins/ui_settings.ts

```diff
@@ -31,15 +32,31 @@ export default function({ getService, getPageObjects }: PluginFunctionalProvider
     });
 
     it('client plugins have access to registered settings', async () => {
-      const settings = await browser.execute('return window.uiSettingsPlugin');
+      const settings = await browser.execute(
+        () => window.np.setup.core.uiSettings.getAll().ui_settings_plugin
+      );
+
       expect(settings).to.eql({
         category: ['any'],
         description: 'just for testing',
         name: 'from_ui_settings_plugin',
         value: '2',
       });
-      const settingsValue = await browser.execute('return window.uiSettingsPluginValue');
+
+      const settingsValue = await browser.execute(() =>
+        window.np.setup.core.uiSettings.get('ui_settings_plugin')
+      );
+
       expect(settingsValue).to.be('2');
+
+      const settingsValueViaObservables = await browser.executeAsync((callback: Function) => {
+        let value;
+        window.np.setup.core.uiSettings.get$('ui_settings_plugin').subscribe(v => (value = v));
+        callback(value);
+        return Promise.resolve(value);
```

Comment:
this is here just to mute TS error. I didn't manage to make it work with a promise. @dmlemeshko FYI

Reply:
It looks like there is a wrong type on the our side
https://github.com/elastic/kibana/blob/master/test/functional/services/browser.ts#L474
Because in Selenium types: 
```
   * Unlike executing synchronous JavaScript with {@link #executeScript},
   * scripts executed with this function must explicitly signal they are
   * finished by invoking the provided callback. This callback will always be
   * injected into the executed function as the last argument, and thus may be
   * referenced with {@code arguments[arguments.length - 1]}. The following
   * steps will be taken for resolving this functions return value against the
   * first argument to the script's callback function:
```